### ISSUE TITLE
Manually added line numbers don't show up if no language set

### ISSUE THREAD

**Issue by crutchcorn** [shikijs/twoslash#155]
I'm currently using `remark-shiki-twoslash` to process code samples. As such, I'd like to add line numbers using CSS, which I can do with the following CSS:

```css
code {
	counter-reset: step;
	counter-increment: step 0;
}

code .line::before {
	content: counter(step);
	counter-increment: step;
	width: 1rem;
	margin-right: 1.5rem;
	display: inline-block;
	text-align: right;
	color: rgba(115,138,148,.4)
}
```

![image](https://user-images.githubusercontent.com/9100169/175160156-bc80d4ca-aad4-4176-b0d6-849b7a9e12e9.png)

This is because the markdown code block looks like this:

![image](https://user-images.githubusercontent.com/9100169/175160215-99e2634c-a434-43ae-bb7b-78f1a42f9931.png)

However, earlier in the markdown code block, I have the following assembly example that I'd like to have line numbers associated with:

![image](https://user-images.githubusercontent.com/9100169/175160296-bde18486-54a5-4e25-bd43-ad246c394584.png)

But it outputs the following in UI:

![image](https://user-images.githubusercontent.com/9100169/175160782-790efd6b-4608-4a34-bc55-df38a3111aca.png)

However, I thought I could work around this by adding a language into the markdown block, but if I change the markdown language to `plain`, I get no lines whatsoever

![image](https://user-images.githubusercontent.com/9100169/175161767-6f1aee73-7ebe-4f2b-b30d-97857982060c.png)

**Comment by orta** [comment]
I'm open to a fix for this 👍🏻

**Comment by crutchcorn** [comment]
Looking into this further (in hopes of making a PR), I think this issue actually:

1) Is caused by the base `shiki` package
2) Is resolved as-of `shiki@0.10`

I think this is the case because, when running my code through a debugger inside of `remark`, I see the following code:

```js
   function codeToThemedTokens(code, lang = 'text', theme, options = { includeExplanation: true }) {
        if (isPlaintext(lang)) {
            return [[{ content: code }]];
        }
        const { _grammer } = getGrammer(lang);
        const { _theme, _colorMap } = getTheme(theme);
        return tokenizeWithTheme(_theme, _colorMap, code, _grammer, options);
    }
```

Where `code` should be split by newlines in order to properly fix this bug.

However, in the new version of the `shiki` codebase the code is as following:

```js
  function codeToThemedTokens(
    code: string,
    lang = 'text',
    theme?: IThemeRegistration,
    options = { includeExplanation: true }
  ) {
    if (isPlaintext(lang)) {
      const lines = code.split(/\r\n|\r|\n/)
      return [...lines.map(line => [{ content: line }])]
    }
    const { _grammar } = getGrammar(lang)
    const { _theme, _colorMap } = getTheme(theme)
    return tokenizeWithTheme(_theme, _colorMap, code, _grammar, options)
  }
```

https://github.com/shikijs/shiki/blob/b0cade2f48f3d6e5c0897c81d0dd06c81f5916ab/packages/shiki/src/highlighter.ts#L119-L122

Which seems to split the lines properly.

Would it be OK for me to make a PR upgrading this dependency despite it's quasi-major release bump?

**Comment by crutchcorn** [comment]
In particular, this seems to've been fixed in `0.10` with this PR:

https://github.com/shikijs/shiki/pull/259
